packages/core-data/src/private-selectors.ts

```diff
@@ -134,6 +139,13 @@ interface SiteData {
 export const getHomePage = createRegistrySelector( ( select ) =>
 	createSelector(
 		() => {
+			const canReadSiteData = select( STORE_NAME ).canUser( 'read', {
```

Comment:
Should we do something similar for `getPostsPageId` and `getTemplateId`

Reply:
Yes for getPostsPageId we should and I included the check. For getTemplateId I think it is not needed as that selector never reads settings directly.